File: example implementations/java/convertor/src/voxel/VoxMatrix.java
package voxel;

public class VoxMatrix {

    public XYZ size;
    public Palette palette;
    public int[][][] content;

    public VoxMatrix(int[][][] content) {
        this.content = content;
        size = new XYZ(content.length, content[0].length, content[0][0].length);
    }

    public VoxMatrix(int[][][] content, Palette palette) {
        this(content);
        this.palette = palette;
    }


    public void add(VoxMatrix matrix) {
        add(new XYZ(), matrix);
    }

    public void add(XYZ offset, VoxMatrix matrix) {
        new XYZ().fromTo(matrix.size, xyz -> {
            int i = matrix.content[xyz.x][xyz.y][xyz.z];
            if (i > 0) {
                xyz = xyz.add(offset);
                content[xyz.x][xyz.y][xyz.z] = i;
            }
        });
    }

    public int get(XYZ xyz) {
        return content[xyz.x][xyz.y][xyz.z];
    }

}

File: example implementations/csharp/cvox-convertor/voxel/VoxMatrix.cs
namespace cvox_convertor.voxel
{
    public class VoxMatrix
    {

        public XYZ size;
        public Palette palette;
        public int[,,] Content;

        public VoxMatrix(int[,,] content)
        {
            Content = content;
            size = new XYZ(content.GetLength(0), content.GetLength(1), content.GetLength(2));
        }

        public VoxMatrix(int[,,] content, Palette palette) : this(content)
        {
            this.palette = palette;
        }

        public void Add(XYZ offset, VoxMatrix matrix)
        {
            new XYZ().FromTo(matrix.size, xyz =>
            {
                int i = matrix.Content[xyz.X, xyz.Y, xyz.Z];
                if (i > 0)
                {
                    xyz += offset;
                    Content[xyz.X, xyz.Y, xyz.Z] = i;
                }
            });
        }

        public int Get(XYZ xyz)
        {
            return Content[xyz.X, xyz.Y, xyz.Z];
        }

    }
}
